src/plugins/fakeNitro.ts

```diff
@@ -392,70 +393,139 @@ export default definePlugin({
+                if (!Array.isArray(child.props.children)) {
+                    child.props.children = modifyChild(child.props.children);
+                    return child;
+                }
+
+                child.props.children = modifyChildren(child.props.children);
+                if (child.props.children.length === 0) return null;
+                return child;
+            }
 
-        const firstContent = newContent[0];
-        if (typeof firstContent === "string") newContent[0] = firstContent.trimStart();
+            return child;
+        };
 
-        return newContent;
+        const modifyChild = (child: ReactElement) => {
+            const newChild = transformChild(child);
+
+            if (newChild?.type === "ul" || newChild?.type === "ol") {
+                this.makeChildrenArray(newChild);
+                if (newChild.props.children.length === 0) return null;
+
+                let listHasAnItem = false;
+                for (const [index, child] of newChild.props.children.entries()) {
+                    if (child == null) {
+                        delete newChild.props.children[index];
+                        continue;
+                    }
+
+                    this.makeChildrenArray(child);
+                    if (child.props.children.length > 0) listHasAnItem = true;
+                    else delete newChild.props.children[index];
+                }
+
+                if (!listHasAnItem) return null;
+
+                newChild.props.children = this.clearEmptyArrayItems(newChild.props.children);
+            }
+
+            return newChild;
+        };
+
+        const modifyChildren = (children: Array<ReactElement>) => {
+            for (const [index, child] of children.entries()) children[index] = modifyChild(child);
+
+            children = this.clearEmptyArrayItems(children);
+            this.trimContent(children);
+
+            return children;
+        };
+
+        try {
+            return modifyChildren(window._.cloneDeep(content));
+        } catch (err) {
+            new Logger("FakeNitro").error(err);
+            return content;
+        }
     },
 
     patchFakeNitroStickers(stickers: Array<any>, message: Message) {
         const itemsToMaybePush: Array<string> = [];
 
         const contentItems = message.content.split(/\s/);
-        if (contentItems.length === 1 && !settings.store.transformCompoundSentence) itemsToMaybePush.push(contentItems[0]);
-        else itemsToMaybePush.push(...contentItems);
+        if (settings.store.transformCompoundSentence) itemsToMaybePush.push(...contentItems);
+        else if (contentItems.length === 1) itemsToMaybePush.push(contentItems[0]);
 
         itemsToMaybePush.push(...message.attachments.filter(attachment => attachment.content_type === "image/gif").map(attachment => attachment.url));
 
         for (const item of itemsToMaybePush) {
+            if (!settings.store.transformCompoundSentence && !item.startsWith("https")) continue;
```

Comment:
```suggestion
            if (!settings.store.transformCompoundSentence && !item.startsWith("http")) continue;
```
don't know the exact context so https might be guaranteed and this suggestion might be obsolete

Reply:
I will do http for safety